Parsers/ASimNetworkSession/Parsers/vimNetworkSessionCiscoMeraki.yaml

```diff
@@ -50,65 +50,413 @@ ParserParams:
+  ];
+  let DvcActionLookup = datatable(pattern: string, DvcAction: string, EventResult: string)[
+      "allow", "Allow", "Success",
+      "deny", "Deny", "Failure",
+      "Blocked", "Deny", "Failure"
+  ];
+  let EventResultLookup = datatable(LogSubType: string, EventResult_type: string)[
+      "association", "Success",
+      "disassociation", "Failure",
+      "Virtual router collision", "Failure",
+  ];
+  let parser=(disabled: bool=false, 
+      starttime: datetime=datetime(null), 
+      endtime: datetime=datetime(null), 
+      eventresult: string='*', 
+      srcipaddr_has_any_prefix: dynamic=dynamic([]),
+      dstipaddr_has_any_prefix: dynamic=dynamic([]),
+      ipaddr_has_any_prefix: dynamic=dynamic([]), 
+      hostname_has_any: dynamic=dynamic([]),
+      dstportnumber: int=int(null),
+      dvcaction: dynamic=dynamic([])
+      ) {
+      let src_or_any = set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+      let dst_or_any = set_union(dstipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+      let allData = union isfuzzy=true
+              (
+              meraki_CL
+              | project-rename LogMessage =  Message
+              ),
+              (
+              Syslog
+              | where Computer in (_ASIM_GetSourceBySourceType('CiscoMeraki'))
+              | project-rename LogMessage =  SyslogMessage
+              );
+      let PreFilteredData = allData
+          | where not(disabled) and (isnull(starttime) or TimeGenerated >= starttime)
+              and (isnull(endtime) or TimeGenerated <= endtime) and (LogMessage has_any("flows", "firewall", "ids-alerts") or LogMessage has_all("security_event", "ids-alerted") or (LogMessage has "events" and (LogMessage has_any ("Blocked DHCP server response", "association") or (LogMessage has "VRRP packet" and not(LogMessage has_any ("VRRP passive", "VRRP active"))) or (LogMessage has "disassociation" and not(LogMessage has_any ("auth_neg_failed", "dhcp"))))) or (LogMessage has "airmarshal_events" and LogMessage has_any("ssid_spoofing_detected", "rogue_ssid_detected")))
+          | extend Parser = extract_all(@"(\d+.\d+)\s([\w\-\_]+)\s([\w\-\_]+)\s([\S\s]+)$", dynamic([1, 2, 3, 4]), LogMessage)[0]
+          | extend
+              LogType = tostring(Parser[2]),
+              Substring = tostring(Parser[3]),
+              Device = tostring(Parser[1])
+          | parse Substring with * "timestamp=" timestamp: string " " *
+          | extend
+              Epoch = iff(isnotempty(timestamp), timestamp, tostring(Parser[0]))
+          | extend
+              EpochTimestamp = split(Epoch, ".")
+          | extend EventStartTime = unixtime_seconds_todatetime(tolong(EpochTimestamp[0]))
+          | extend EventEndTime = EventStartTime
+          | where  (isnull(dstportnumber)) or Substring has tostring(dstportnumber)
+              and (array_length(dvcaction) == 0 or LogMessage has_any (dvcaction))
+              and array_length(hostname_has_any) == 0;
```

Comment:
hostname_has_any filter isn't working. Please fix this condition.

Reply:
Done